scripts/superdesk/editor2/add-embed.ctrl.js

```diff
@@ -55,21 +53,81 @@ function SdAddEmbedController (embedService, $element, $timeout, $q, _, EMBED_PR
                         };
                     });
                 });
-            // otherwise we use the content of the field directly
-            } else {
-                var embedType = EMBED_PROVIDERS.custom;
+            }
+            function parseRawEmbedCode() {
+                var waitFor = [];
+                var embedBlock = {
+                    body: vm.input,
+                    provider: EMBED_PROVIDERS.custom
+                };
+                var providersKnown = [
+                    {
+                        pattern: /twitter\.com\/widgets\.js/g,
+                        name: EMBED_PROVIDERS.twitter
+                    },
+                    {
+                        pattern: /www\.youtube\.com/g,
+                        name: EMBED_PROVIDERS.youtube
+                    },
+                    {
+                        pattern: /src="(.*vidible\.tv.*pid=.*.js)/g,
+                        name: EMBED_PROVIDERS.vidible,
+                        callback: function(match) {
+                            var url = 'https://' + match[1];
+                            // $http raise a Same Origin Policy error, so we use jquery here
```

Comment:
I don't know why $http raises this error and jquery doesn't... maybe `jquery.ajax` change the `origin` value.